src/ProjectOrigin.Vault/Repositories/WithdrawnCursorRepository.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading.Tasks;
+using Dapper;
+using ProjectOrigin.Vault.Extensions;
+using ProjectOrigin.Vault.Models;
+
+namespace ProjectOrigin.Vault.Repositories;
+
+public interface IWithdrawnCursorRepository
+{
+    Task<IEnumerable<WithdrawnCursor>> GetWithdrawnCursors();
+    Task InsertWithdrawnCursor(WithdrawnCursor withdrawnCursor);
+    Task UpdateWithdrawnCursor(WithdrawnCursor withdrawnCursor);
+}
+
+public class WithdrawnCursorRepository : IWithdrawnCursorRepository
+{
+    private readonly IDbConnection _connection;
+
+    public WithdrawnCursorRepository(IDbConnection connection) => this._connection = connection;
+
+    public async Task<IEnumerable<WithdrawnCursor>> GetWithdrawnCursors()
+    {
+        using var gridReader = await _connection.QueryMultipleAsync("SELECT * FROM withdrawn_cursors");
```

Comment:
Why use QueryMultiple? instead of just .Query<WithdrawnCursor> ?

Reply:
that works too. corrected